test/forms/labels/input-label.test.js

```diff
@@ -1,93 +1,109 @@
 import React from 'react'
-import { mount } from 'enzyme'
 import { InputLabel } from '../../../src/'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 
 const name = 'person.firstName'
 const formattedName = 'First Name'
 
 test('when label is false - does not render a label', () => {
-  const wrapper = mount(<InputLabel name={name} label={false} />)
-  expect(wrapper.find('label').exists()).toEqual(false)
+  render(<InputLabel name={name} label={false} />)
+  expect(screen.queryByText(formattedName)).not.toBeInTheDocument()
```

Comment:
I'm not sure this test works as intended. It still would pass if a label was rendered. It just tests the variable, which is never passed anyway.

Reply:
What do you mean it just tests the variable? When I remove `label={false}` the test fails

![](https://p-1MFmr4.b2.n0.cdn.getcloudapp.com/items/8LuYXx7P/5c8c9d61-4c45-4649-be6b-3e3be406379a.jpg?v=d42329fd071713c07727d64f18c4cc42)

Reply:
Oh I just noticed that `InputLabel` converts the text using `convertNameToLabel` internally. So that's why you're testing against this variable.

Reply:
Testing against `formattedName` didn't make sense to me because you never passed this value anyway. I see the connection now.

Reply:
Yes that is true, if you passed in a different label the test would need to be updated to search for that label

Reply:
Anyway, since you're strictly testing for `label={false}` in this test, I think we're fine.